### PR TITLE
rename ui-form into form and moved  files to utils

### PR DESCRIPTION
**Pull request by phpgeekfr**

### CODE REVIEW

File: libs/utils/src/lib/form/entity.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { FormControl, FormGroup, FormArray } from "@angular/forms";
+import { rules } from "./validators";
```
Comment (GrandSchtroumpf) [inline libs/utils/src/lib/form/entity.ts]: Should be inside a "validators" folder. At the same level of "form"

File: libs/utils/src/lib/form/entity.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+// Generic Control 
+export type EntityControl<T> = {
+  [key in keyof T]: FormControl | FormArray
```
Comment (GrandSchtroumpf) [inline libs/utils/src/lib/form/entity.ts]: Could also be "| FormGroup".

File: libs/utils/src/lib/form/entity.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+// Generic Entities
+export class EntityForm<T> extends FormGroup {
+  constructor(controls: EntityControl<T>, validators?: any) {
+    super(controls, validators)
+  }
+}
+
+export class EntityRulesForm<T> extends EntityForm<T> {
+  public rules = rules;
```
Comment (GrandSchtroumpf) [inline libs/utils/src/lib/form/entity.ts]: Maybe call that "validators"

File: libs/account/src/lib/profile/profile-edit/profile-edit.form.ts [hunk 4]
```diff
@@ -1,35 +1,31 @@
...
-  protected form : AbstractFormControls;
-
-  constructor(data?: any, controls? : any, validators?: any ) {
-    const f = new ProfileFormControls(data);
-    super(
-      controls !== undefined ? controls : f.controls,
-      validators !== undefined ? validators : f.validators
-    );
-    this.form = f;
+function createControls(entity: Profile): EntityControl<Profile> {
```
Comment (GrandSchtroumpf) [inline libs/account/src/lib/profile/profile-edit/profile-edit.form.ts]: Shouldn't it be `createProfileControls`?
Maybe you can have a `createProfile` method that will take care of filling all empty params : 
```
function createProfile(params?: Partial<Profile>): Profile {
  return {
    email: '',
    ...,
    ...(params || {})
  } as Profile
}
```
like that you can do :
```
function createProfileControls(entity: Partial<Profile>): EntityControl<Profile> {
  const profile = createProfile(entity);
  return {
    email: new EmailControl(profile.email, true)
  }
}
```

**Comment by phpgeekfr** [comment]
closed by mistake

**Pull request merged**